reference-ril/reference-ril.c

```diff
@@ -3299,12 +4693,18 @@ mainLoop(void *param __unused)
             if (s_port > 0) {
                 fd = socket_loopback_client(s_port, SOCK_STREAM);
             } else if (s_device_socket) {
+                if (!s_client_id) {
+                    s_client_id = "";
+                }
+
                 if (!strcmp(s_device_path, "/dev/socket/qemud")) {
                     /* Before trying to connect to /dev/socket/qemud (which is
                      * now another "legacy" way of communicating with the
                      * emulator), we will try to connecto to gsm service via
                      * qemu pipe. */
-                    fd = qemu_pipe_open("qemud:gsm");
+                    char buffer[32];
+                    snprintf(buffer, sizeof(buffer), "qemud:gsm%s", s_client_id);
```

Comment:
https://github.com/vicamo/b2g_platform_hardware_ril/blame/next/b2g/b2g-lollipop-mr1/rild/rild.c#L143-L145 gives a default value to `clientId` (d365219a1873e42d2ff669e13a9fe6ce6973bd51), the first rild tries to open pipe __qemu:gsm0__ which makes the it unable to communicate with qemu.